package/MDAnalysis/core/selection.py

```diff
@@ -537,6 +537,101 @@ def _apply(self, group):
+    combination = [
+        "x",
+        "y",
+        "z",
+        "xy",
+        "xz",
+        "yz",
+        "yx",
+        "zx",
+        "zy",
+        "xyz",
+        "xzy",
+        "yxz",
+        "yzx",
+        "zxy",
+        "zyx",
+    ]
+    axis_map = ["x", "y", "z"]
+
+    def __init__(self, parser, tokens):
+        super().__init__(parser, tokens)
+        self.periodic = parser.periodic
+        self.direction = tokens.popleft()
+        if self.direction not in self.combination:
+            raise ValueError(
+                "The direction '{}' is not valid. Must be combination of {}"
+                "".format(self.direction, list(self.index_map.keys()))
+            )
+        else:
+            for d in self.direction:
+                setattr(self, "{}max".format(d), float(tokens.popleft()))
+                setattr(self, "{}min".format(d), float(tokens.popleft()))
+        self.sel = parser.parse_expression(self.precedence)
+
+    @return_empty_on_apply
+    def _apply(self, group):
+        sel = self.sel.apply(group)
+        if len(sel) == 0:
+            return group[[]]
+        # Calculate vectors between point of interest and our group
+        vecs = group.positions - sel.center_of_geometry()
+        range_map = {}
+
+        for d in self.direction:
+            axis_index = self.index_map.get(d)
+            axis_max = self.__getattribute__("{}max".format(d))
+            axis_min = self.__getattribute__("{}min".format(d))
+            range_map[axis_index] = (axis_min, axis_max)
+
+        if self.periodic and group.dimensions is not None:
+            box = group.dimensions[:3]
+
+            for k, v in range_map.items():
+                axis_index = k
+                axis_min, axis_max = v[0], v[1]
+
+                axis_height = axis_max - axis_min
+                if axis_height > box[axis_index]:
+                    raise NotImplementedError(
+                        "The total length of the box selection in {} ({:.3f}) "
+                        "is larger than the unit cell's {} dimension ({:.3f}). "
+                        "Can only do selections where it is smaller or equal."
+                        "".format(
+                            self.axis_map[axis_index],
+                            axis_height,
+                            self.axis_map[axis_index],
+                            box[axis_index],
+                        )
+                    )
+
+            if np.all(group.dimensions[3:] == 90.0):
+                # Orthogonal version
+                vecs -= box[:3] * np.rint(vecs / box[:3])
+            else:
+                # Triclinic version
+                tribox = group.universe.trajectory.ts.triclinic_dimensions
+                vecs -= tribox[2] * np.rint(vecs[:, 2] / tribox[2][2])[:, None]
+                vecs -= tribox[1] * np.rint(vecs[:, 1] / tribox[1][1])[:, None]
+                vecs -= tribox[0] * np.rint(vecs[:, 0] / tribox[0][0])[:, None]
```

Comment:
this isn't technically correct for some triclinic boxes with some shifts.

Reply:
I think you'll want this function: https://github.com/MDAnalysis/mdanalysis/blob/c36313e4e849bc37033a142454e9e593db2b3272/package/MDAnalysis/lib/distances.py#L1791

But it won't handle some, but not all, box dimensions being given (i.e. semi periodic calculation)